workflows/plugins/include/utils/workflows.py

```diff
@@ -123,3 +125,51 @@ def post_pdf_to_grobid(workflow_id, workflow, s3_hook, bucket_name):
         response = grobid_http_hook.call_api("api/processHeaderDocument", files=files)
 
     return response
+
+
+def get_source_for_root(source):
+    """Source for the root workflow object.
+    Args:
+        source(str): the record source.
+    Return:
+        (str): the source for the root workflow object.
+    Note:
+        For the time being any workflow with ``acquisition_source.source``
+        different than ``arxiv`` and ``submitter`` will be stored as
+        ``publisher``.
+    """
+    return source if source in ["arxiv", "submitter"] else "publisher"
+
+
+def read_wf_record_source(record_uuid, source):
+    """Retrieve a record from the ``WorkflowRecordSource`` table.
+    Args:
+        record_uuid(uuid): the uuid of the record
+        source(string): the acquisition source value of the record
+    Return:
+        (dict): the given record, if any or None
+    """
+    if not source:
+        return
+
+    source = get_source_for_root(source)
+
+    inspire_http_hook = InspireHttpHook()
+
+    try:
+        response = inspire_http_hook.call_api(
+            endpoint="api/literature/workflows_record_sources",
```

Comment:
this assumes we are going to keep using the WorkflowsRecordSources from hep. I guess thats how we are going to proceed , but to be validated